pkg/puller/puller_test.go

```diff
@@ -26,47 +24,35 @@ import (
 	"github.com/ethersphere/bee/pkg/util/testutil"
 )
 
-const max = math.MaxUint64
-
-var (
-	call = func(b uint8, f, t uint64) c {
-		return c{b: b, f: f, t: t}
-	}
-
-	reply = mockps.NewReply // alias to make code more readable
-)
-
 // test that adding one peer starts syncing
 func TestOneSync(t *testing.T) {
 	t.Parallel()
 
 	var (
-		addr        = test.RandomAddress()
-		cursors     = []uint64{1000, 1000, 1000}
-		liveReplies = []uint64{1001}
+		addr    = test.RandomAddress()
+		cursors = []uint64{1000, 1000, 1000}
+		replies = []mockps.SyncReply{
+			{Bin: 1, Start: 1, Topmost: 1000, Peer: addr},
+			{Bin: 2, Start: 1001, Topmost: 1001, Peer: addr}}
 	)
 
-	bsMock.New(bsMock.WithReserveState(&postage.ReserveState{StorageRadius: 1}))
-
-	puller, _, kad, pullsync := newPuller(t, opts{
+	_, _, kad, pullsync := newPuller(t, opts{
```

Comment:
So in the tests I dont see pullSync being closed at the end?...

Reply:
https://github.com/ethersphere/bee/pull/3865/files#diff-b71ab66e6a2b746950bdff08673d4e179be7fd002879d3002d926db343e5f6dcR537